frontend/src/components/eMIB/CollapsingItemContainer.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import "../../css/collapsing-item.css";
+
+const styles = {
+  container: {
+    position: "relative"
+  },
+  button: {
+    width: "100%",
+    textAlign: "left"
+  },
+  envelopeIcon: {
+    marginRight: 6
+  },
+  collapsingIcon: {
+    position: "absolute",
+    top: 0,
+    right: 0,
+    margin: "11px 24px 0 0",
+    color: "#00565E",
+    pointerEvents: "none"
+  },
+  contentContainer: {
+    margin: 12
+  }
+};
+
+export const ICON_TYPE = {
+  email: "fas fa-envelope",
+  task: "fas fa-tasks"
+};
+
+class CollapsingItemContainer extends Component {
+  static propTypes = {
+    iconType: PropTypes.string,
+    title: PropTypes.string.isRequired,
+    body: PropTypes.object.isRequired
+  };
+
+  state = {
+    isHidden: true,
+    buttonClass: "btn btn-secondary",
+    iconClass: "fas fa-angle-down",
+    containerClass: ""
+  };
+
+  expandItem = () => {
+    if (this.state.isHidden) {
+      this.setState({
+        isHidden: false,
+        buttonClass: "btn btn-primary expanded-button-style",
+        iconClass: "fas fa-angle-up expand-icon-style",
+        containerClass: "expanded-container-style"
+      });
+    } else {
+      this.setState({
+        isHidden: true,
+        buttonClass: "btn btn-secondary",
+        iconClass: "fas fa-angle-down",
+        containerClass: ""
+      });
+    }
+  };
+
+  render() {
+    const { iconType, title, body } = this.props;
```

Comment:
It'd be better to just store the isHidden in state, and then in render you can have some conditionals to determine what classes to use. 

State typically doesn't hold this render-specific information, and it allows all the render specific information to happen in the render function.

Reply:
Fixed in commits #https://github.com/code-for-canada/project-thundercat/pull/171/commits/4a68efd4d3757b2d5d66d003b9a876b04c60718f and #https://github.com/code-for-canada/project-thundercat/pull/171/commits/aa2985989718ea9a524ef1b9156a2099f858b2af (See PR https://github.com/code-for-canada/project-thundercat/pull/171/)

Is that what you had in mind?

Reply:
Yeah that's exactly what I had in mind.